File: app/src/main/java/com/mxt/anitrend/base/interfaces/base/PreferenceConverter.java
package com.mxt.anitrend.base.interfaces.base;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * Created by max on 2018/09/01.
 * Convert objects to json values and back.
 */
public interface PreferenceConverter<T> {

    @NonNull
    T convertToEntity(@Nullable String json);

    @NonNull
    String convertToJson(@Nullable T entity);
}

File: app/src/main/java/com/mxt/anitrend/base/interfaces/base/PreferenceConverter.kt
package com.mxt.anitrend.base.interfaces.base

/**
 * Created by max on 2018/09/01.
 * Convert objects to json values and back.
 */
interface PreferenceConverter<T> {
    fun convertToEntity(json: String?): T

    fun convertToJson(entity: T?): String
}
